### PR TITLE
[fuchsia] remove obsolete input-related integration tests

### PR DESCRIPTION
**Pull request by caroqliu**
This PR removes mouse-input and text-input test suites. Fixes: fxbug.dev/125303

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] ~I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.~ N/a as this change only removes code.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/fuchsia/flutter/tests/integration/touch-input/touch-input-test.cc [hunk 1]
```diff
@@ -492,28 +492,6 @@ TEST_P(FlutterEmbedTapTest, FlutterEmbedTap) {
   ASSERT_EQ(touch_injection_request_count(), 2);
 }
 
-TEST_P(FlutterEmbedTapTest, FlutterEmbedHittestDisabled) {
-  FML_LOG(INFO) << "Initializing scene";
-  AddComponentArgument("--no-hitTestable");
```
Comment (arbreng) [inline shell/platform/fuchsia/flutter/tests/integration/touch-input/touch-input-test.cc:497]: We should delete the vestigial code that this is no longer exercising:  This means removing all code in https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/tests/integration/touch-input/embedding-flutter-view/lib/embedding-flutter-view.dart which references "hitTestable".

**Review by arbreng — APPROVED** [review]
Please delete the vestigial code that these tests no longer exercise.  It's fine to do that in a follow-up CL but let's please do it soon.

Its important for the flutter code to be as minimal as possible going forward, and to have test coverage on what's left.

For mouse, this is everything related to MouseSource.  Start here: https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/platform_view.cc#L47  There is also the mouse-related code in PointerDelegate: https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/pointer_delegate.cc Keep the touch-related parts of PointerDelegate.

For keyboard, that is everything related to fuchsia::ui::input3::Keybaord.  Start here https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/platform_view.cc#L45.  Keep the IME-related parts of TextDelegate.

For hit-test, that is all code https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/tests/integration/touch-input/embedding-flutter-view/lib/embedding-flutter-view.dart which references "hitTestable".  It is also all code in the C++ which references hit_testable.  Start here: https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/platform_view.cc#L466

Let's talk about this in chat or a meeting, I can give you many more details.

**Comment by caroqliu** [comment]
> Please delete the vestigial code that these tests no longer exercise. It's fine to do that in a follow-up PR but let's please do it soon. I'd love if you just went ahead and did it in this PR.

This is a good point. Since removing these tests isn't urgent, I'll go ahead and make the removals of the code under test in this same PR and re-request review when that's ready.

> For hit-test, that is all code https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/tests/integration/touch-input/embedding-flutter-view/lib/embedding-flutter-view.dart which references "hitTestable". It is also all code in the C++ which references hit_testable. Start here: https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/platform_view.cc#L466

Thanks for the reference for hit-test, too. On second thought, I think it's valuable for this code path to be tested until the Flatland test ui stack is the only variant being supported, which is at least a couple months away. I've restored the test in this change and modified #41118 to run it on GFX Scene Manager.

> Let's talk about this in chat or a meeting, I can give you many more details.

Anything you still wanted to cover?

**Comment by chinmaygarde** [comment]
Are we close to landing this? If not, you can mark it as WIP so we don't bother you in triage :)

**Comment by caroqliu** [comment]
Closing this PR for now since we are rescheduling this work.

**Pull request closed without merging**